### PR TITLE
hasura-cli 2.0.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by cho-m** [comment]
Looks like there is an extra binary that needs to be built and then copied into directory
- Node-based `cli-ext`: https://github.com/hasura/graphql-engine/tree/master/cli-ext#building-the-binaries
- Copy step: https://github.com/hasura/graphql-engine/blob/v2.0.1/cli/Makefile#L57-L58
    ```make
    copy-cli-ext: clean-cli-ext
	    cp -r $(CLI_EXT_BIN_DIR) ./internal/cliext/bin
    ```
- Build step: https://github.com/hasura/graphql-engine/blob/v2.0.1/cli/Makefile#L24-L35
    ```make
    build: export CGO_ENABLED=0
    build: copy-cli-ext
    build:
    ifndef HAS_GOX
	    cd ~ && go get github.com/mitchellh/gox && cd -
    endif
	    gox -ldflags '-X github.com/hasura/graphql-engine/cli/v2/version.BuildVersion=$(VERSION) -X github.com/hasura/graphql-engine/cli/v2/plugins.IndexBranchRef=$(PLUGINS_BRANCH) -s -w -extldflags "-static"' \
	    -rebuild \
	    -os="$(OS)" \
	    -arch="amd64" \
	    -output="$(OUTPUT_DIR)/$(VERSION)/cli-hasura-{{.OS}}-{{.Arch}}" \
	    ./cmd/hasura/
    ```

Makefile probably shouldn't be used as-is since it automatically installs `gox` and also locks architecture to `amd64`.

**Comment by chenrui333** [comment]
I see, looking into the license assets now.

Also there build args for version is updated to use v2 as well.

**Comment by cho-m** [comment]
`make copy-cli-ext` fails as it tries CI-specific location `cp -r "/build/_cli_ext_output" ./internal/cliext/bin`

Probably just directly copy it.

**Comment by chenrui333** [comment]
Interesting, it actually passed my local build 🤔

**Comment by cho-m** [comment]
Should be due to the Makefile conditional. If run locally, it will use `else` and find relative directory:
```make
ifeq (${CI}, true)
    CLI_EXT_BIN_DIR := "/build/_cli_ext_output"
else
    CLI_EXT_BIN_DIR := "../cli-ext/bin"
endif
```

**Comment by chenrui333** [comment]
Where did we define `CI`, let me see how it goes with explicitly specifying `CI=false`

**Comment by chenrui333** [comment]
Looks like the CI env config change fixed the issue.

File: Formula/hasura-cli.rb [hunk 1]
```diff
@@ -14,22 +15,27 @@ class HasuraCli < Formula
...
+    ].join(" ")
 
     cd "cli" do
-      system "go", "build", *std_go_args, "-ldflags", ldflags.join(" "), "-o", bin/"hasura", "./cmd/hasura/"
-
-      system bin/"hasura", "completion", "bash", "--file", "completion_bash"
-      bash_completion.install "completion_bash" => "hasura"
+      ENV["CI"] = "false"
+      system "make", "build-cli-ext"
+      system "make", "copy-cli-ext"
```
Comment (carlocab) [inline Formula/hasura-cli.rb:32]:
```suggestion
      with_env(CI: "false") do
        system "make", "build-cli-ext"
        system "make", "copy-cli-ext"
      end
```
Let's try to preserve the environment where we can.

Comment (chenrui333) [reply Formula/hasura-cli.rb]: 👍

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**